### PR TITLE
[SPARK-15901] [SQL] [TEST] Verification of CONVERT_METASTORE_ORC and CONVERT_METASTORE_PARQUET

### PR DESCRIPTION
**Pull request by gatorsmile**
#### What changes were proposed in this pull request?

So far, we do not have test cases for verifying whether the external parameters `HiveUtils .CONVERT_METASTORE_ORC` and `HiveUtils.CONVERT_METASTORE_PARQUET` properly works when users use non-default values. This PR is to add such test cases for avoiding potential regression.
#### How was this patch tested?

N/A

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/parquetSuites.scala [hunk 1]
```diff
@@ -676,6 +676,46 @@ class ParquetSourceSuite extends ParquetPartitioningTest {
...
+                fail(s"Expecting no conversion from parquet to data sources, " +
+                  s"but got:\n$queryExecution")
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
```
Comment (clockfly) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/parquetSuites.scala]: It is a bit costy to write a end to end test like this, how about adding an Unit test using SimpleAnalyzer?

Comment (gatorsmile) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/parquetSuites.scala]: Thank you for your review! @clockfly 

I like your suggestion, but the conf `CONVERT_METASTORE_ORC` and `CONVERT_METASTORE_PARQUET` are not available in `CatalystConf` and `SimpleCatalystConf`. I am hesitant to change the source codes for improving test case coverage. 

Actually, I am not sure if we should keep them in the `Analyzer` in the future release. These two rules are not used for resolution. Instead, it is for performance enhancement. Ideally, I think they are `Optimizer` rules. 

cc @cloud-fan @liancheng

**Comment by liancheng** [comment]
LGTM, merging to master and branch-2.0. Thanks!

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/parquetSuites.scala (hunk 1 above)
Comment (liancheng) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/parquetSuites.scala]: Having Parquet/ORC conversions in the optimizer is also reasonable, but doesn't seem to bring much benefits. Furthermore, it also requires us to refactor the optimizer to support Hive specific rules.

**Pull request closed without merging**